### PR TITLE
add JobSet periodic release and main tests

### PR DESCRIPTION
**Pull request by omerap12**
issue: https://github.com/kubernetes-sigs/jobset/issues/524

### CODE REVIEW

File: config/jobs/kubernetes-sigs/jobset/jobset-presubmit-release-0.4.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        args:
+        - test-integration
+        resources:
+          limits:
+            cpu: 3
+            memory: 10Gi
+          requests:
+            cpu: 3
+            memory: 10Gi
+  - name: pull-jobset-test-e2e-main-1-27
```
Comment (kannon92) [inline config/jobs/kubernetes-sigs/jobset/jobset-presubmit-release-0.4.yaml:62]:
```suggestion
  - name: pull-jobset-test-e2e-release-0.4-1-27
```

let’s make sure the e2e jobs match the release branch and not main

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-presubmit-release-0.4.yaml]: Yeap

**Review by kannon92 — COMMENTED** [review]
/retitle add JobSet periodic release and main tests

**Review by kannon92 — COMMENTED** [review]
/ok-to-test

/assign @danielvegamyhre

**Comment by kannon92** [comment]
Check linter. You may have duplicate names or indentation issues

**Comment by omerap12** [comment]
> Check linter. You may have duplicate names or indentation issues

Yes, noticed. I'll get home and fix this. Thanks :)

**Comment by omerap12** [comment]
Hey @kannon92  @danielvegamyhre  , 
any update?

**Comment by danielvegamyhre** [comment]
Thanks @omerap12 I will take a look at this tomorrow

**Comment by omerap12** [comment]
> @omerap12 did you use any existing tests as a reference when building this? If so can you share them

Yes, this is the presubmit tests for jobset:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml

So I implemented those tests with reference to kueue:
https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/kueue

File: config/jobs/kubernetes-sigs/jobset/jobset-presubmit-release-0.4.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          value: kindest/node:v1.27.3
+        - name: BUILDER_IMAGE
+          value: public.ecr.aws/docker/library/golang:1.21
+        command:
+        # generic runner script, handles DIND, bazelrc for caching, etc.
+        - runner.sh
+        args:
+        - make
+        - kind-image-build
+        - test-e2e
```
Comment (danielvegamyhre) [inline config/jobs/kubernetes-sigs/jobset/jobset-presubmit-release-0.4.yaml:89]: In JobSet the make rule for e2e tests is [test-e2e-kind](https://github.com/kubernetes-sigs/jobset/blob/947df5da269dadd66a3c819497983834a834f586/Makefile#L27). Please check the various make commands in this PR to ensure they are using the JobSet make rules and not the Kueue ones.

File: config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml [hunk 3]
```diff
@@ -0,0 +1,190 @@
+periodics:
```
Comment (danielvegamyhre) [inline config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml:1]: I thought we were just adding periodic tests on the release branches, what are the other presubmit files about?

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml:1]: I thought of adding presubmit checks for release branches as well. I can remove them if you'd like

Comment (danielvegamyhre) [reply config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml]: Would the presubmits only trigger when we are merging a PR into the given release branch? Or does it run unconditionally?

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml]: Presubmit jobs runs for pull requests into the given branch.  periodic jobs run unconditionally

File: config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      containers:
+      - image: public.ecr.aws/docker/library/golang:1.22
+        command:
+        - make
+        args:
+        - test-integration
+        resources:
+          limits:
+            cpu: 3
+            memory: 10Gi
```
Comment (danielvegamyhre) [inline config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:67]: 10Gi seems like a lot, is this what Kueue uses? I don't think we need that much, see jobset controller manager resource requests/limits [here](https://github.com/kubernetes-sigs/jobset/blob/235745c5ed1bd70f54f36aa8782122e3326a04b6/config/components/manager/manager.yaml#L94-L100)

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:67]: Yes, kueue using 10Gib too , see [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml#L95)

And the resources described here is not for the jobset controller manager, its for this container [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kueue/kueue-presubmits-release-0-5.yaml#L76)

Comment (danielvegamyhre) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:67]: I see, ok let's stick with 3 for now and see how that performs, we can bump it if necessary.

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:67]: Sure 👍

File: config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        - name: E2E_KIND_VERSION
+          value: kindest/node:v1.27.3
+        - name: BUILDER_IMAGE
+          value: public.ecr.aws/docker/library/golang:1.22
+        command:
+        # generic runner script, handles DIND, bazelrc for caching, etc.
+        - runner.sh
+        args:
+        - make
+        - kind-image-build
```
Comment (danielvegamyhre) [inline config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:101]: kind-image-build is already ran as part of `test-e2e-kind` rule, we can remove it here.

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-release-0-4.yaml:101]: Got it. fixing right now

File: config/jobs/kubernetes-sigs/jobset/jobset-periodics-main.yaml [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        - name: GO_TEST_FLAGS
+          value: "-race -count 3"
+        command:
+        # generic runner script, handles DIND, bazelrc for caching, etc.
+        - runner.sh
+        args:
+        - make
+        - test
+        securityContext:
+          privileged: true
```
Comment (danielvegamyhre) [inline config/jobs/kubernetes-sigs/jobset/jobset-periodics-main.yaml:32]: any idea why this is necessary?

Comment (omerap12) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-main.yaml:32]: Because you are running Docker in Docker I think it is necessary

Comment (danielvegamyhre) [reply config/jobs/kubernetes-sigs/jobset/jobset-periodics-main.yaml:32]: Gotcha, ok

File: config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml (hunk 3 above)
Comment (danielvegamyhre) [reply config/jobs/kubernetes-sigs/jobset/jobset-release-blocking.yaml]: Ok perfect

**Comment by danielvegamyhre** [comment]
/lgtm
/approve

Thanks @omerap12!

**Comment by danielvegamyhre** [comment]
@omerap12 some of the tests added here are failing ([example](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-jobset-test-e2e-release-0-4-1-28/1790345434540019712)) can you take a look please?

**Comment by omerap12** [comment]
> @omerap12 some of the tests added here are failing ([example](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-jobset-test-e2e-release-0-4-1-28/1790345434540019712)) can you take a look please?

sure, Ill take a look

**Comment by omerap12** [comment]
> @omerap12 some of the tests added here are failing ([example](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-jobset-test-e2e-release-0-4-1-28/1790345434540019712)) can you take a look please?

Found the problem. Ill create another PR to fix this soon.

**Comment by alculquicondor** [comment]
Why use `kubekins-e2e` for release-0.5, if the main branch uses just `golang`? https://github.com/kubernetes/test-infra/blob/3c1c806d19d680b83e4f0a128f1d38641607d2ea/config/jobs/kubernetes-sigs/jobset/jobset-presubmit-main.yaml#L46

**Comment by omerap12** [comment]
> Why use `kubekins-e2e` for release-0.5, if the main branch uses just `golang`? https://github.com/kubernetes/test-infra/blob/3c1c806d19d680b83e4f0a128f1d38641607d2ea/config/jobs/kubernetes-sigs/jobset/jobset-presubmit-main.yaml#L46

I'll take a look and create a PR soon

**Pull request merged**